test/qa/test_connect.py

```diff
@@ -483,52 +483,57 @@ def test_status_disconnected(tech, proto, obfuscated):
+
         connect_time = int(time.time())
 
-        assert hostname in sh.nordvpn.status()
+        data_10MB = requests.get("http://speedtest.tele2.net/10MB.zip")
+        destination_file = "file_10MB"
+        if data_10MB.status_code == 200:
+            with open(destination_file, "wb") as file:
+                file.write(data_10MB.content)
 
-        assert socket.gethostbyname(hostname) in sh.nordvpn.status()
+        files = {'file': open(destination_file, 'rb')}
+        requests.post("http://speedtest.tele2.net/upload.php", files=files)
+
+        current_time = int(time.time())
+
+        status_output = sh.nordvpn.status().replace('\r-\r  \r\r-\r  \r','')
+        status_info = dict((a.strip().lower(), b.strip())  
+            for a, b in (element.split(':') 
+                for element in filter(lambda line: len(line.split(':')) == 2, status_output.split('\n')))) 
+        
+        assert "Connected" in status_info['status']
 
-        assert "Connected" in sh.nordvpn.status()
+        assert hostname in status_info['hostname']
+
+        assert socket.gethostbyname(hostname) in status_info['ip']
 
         city, country = server.get_server_info(name)
-        assert city in sh.nordvpn.status()
-        assert country in sh.nordvpn.status()
+        assert country in status_info['country']
+        assert city in status_info['city']
 
-        assert tech.upper() in sh.nordvpn.status()
+        assert tech.upper() in status_info['current technology']
 
         if tech == "openvpn":
-            assert proto.upper() in sh.nordvpn.status()
+            assert proto.upper() in status_info['current protocol']
         else:
-            assert "UDP" in sh.nordvpn.status()
-
-        data_10MB = requests.get("http://212.183.159.230/10MB.zip")
-        destination_file = "file_10MB"
-        if data_10MB.status_code == 200:
-            with open(destination_file, "wb") as file:
-                file.write(data_10MB.content)
+            assert "UDP" in status_info['current protocol']
 
-        files = {'file': open(destination_file, 'rb')}
-        requests.post("http://speedtest.tele2.net/upload.php", files=files)
-        
-        transfer = str(sh.nordvpn.status()).split("\n")[7]
-        transfer_received = float(transfer.split(" ")[1])
-        transfer_sent = float(transfer.split(" ")[4])
+        transfer_received = float(status_info['transfer'].split(" ")[0])
+        transfer_sent = float(status_info['transfer'].split(" ")[3])
 
         #The app shows transfer size in MiB, 10MB = ~10.49MiB
         assert transfer_received > 10 and transfer_received < 12
         assert transfer_sent > 10 and transfer_sent < 12
 
-        uptime = str(sh.nordvpn.status()).split("\n")[8]
-        time_connected = int(uptime.split(" ")[1])
-        current_time = int(time.time())
+        time_connected = int(status_info['uptime'].split(" ")[0])
         time_passed = current_time - connect_time
```

Comment:
This might not be problematic because:
* if `uptime` is bigger then 1 minute, the first component will be minutes instead of seconds, and then it needs to be `* 60`
* if it is still in seconds, the diff will not return the uptime since it was connected, but some time before the connection time. Adding the `uptime` might give the correct value.

Previous check might have been better, if instead of `time.time()` `time.monotonic`(https://docs.python.org/3/library/time.html#time.monotonic) was used.

Reply:
Changed to `time.monotonic` and for the issue that would come up if connection time was more than a minute came up with such workaround:

```
time_connected = int(status_info['uptime'].split(" ")[0])
time_connected_seconds = int(status_info['uptime'].split(" ")[2]) if "minute" in status_info["uptime"] else 0
time_passed = status_time - connect_time
     
if "minute" in status_info["uptime"]:
            assert time_connected * 60 + time_connected_seconds >= time_passed - 1 and time_connected * 60 + time_connected_seconds <= time_passed + 1
else:    
            assert time_connected >= time_passed - 1 and time_connected <= time_passed + 1 
```
            
To test this, temporarily added `time.sleep(50)` before executing `sh.nordvpn.status()` and tests pass (also to double-check, printed values in logs to see if everything is correct and matching) 

Do you think this look okay or is there a better way to approach this?

Reply:
An example from logs then:

DBG: 1 // `time_connected` from status (1 minute)
DBG: 19 // `time_connected_seconds`
DBG: 79.82627920698724 // `time_passed`
DBG: {'status': 'Connected', 'hostname': 'us5063.nordvpn.com', 'ip': '185.245.87.59', 'country': 'United States', 'city': 'Los Angeles', 'current technology': 'NORDLYNX', 'current protocol': 'UDP', 'transfer': '11.50 MiB received, 11.66 MiB sent', 'uptime': '1 minute 19 seconds'} //The output of status

Reply:
Looks good. :+1: 
if you want, you can move `time_connected_seconds` inside the `if "minute" in status_info["uptime"]:` and you can write just `time_connected_seconds = int(status_info['uptime'].split(" ")[2])`.

Reply:
Just my opinion, but I don't think it was necessary to bother with the "minute" case, this test is never going to take a full minute :D

Reply:
True, but oh well, just in case :D